arch/cpu/nrf/net/nrf-ieee-driver-arch.c

```diff
@@ -720,10 +719,18 @@ read_frame(void *buf, unsigned short bufsize)
 
   /* Latch timestamp values for this most recently received frame */
   timestamps.phr = rx_buf.phr;
-  timestamps.framestart = nrf_timer_cc_get(NRF_TIMER0, NRF_TIMER_CC_CHANNEL3);
-  timestamps.end = nrf_timer_cc_get(NRF_TIMER0, NRF_TIMER_CC_CHANNEL2);
+  timestamps.framestart = nrf_timer_cc_get(NRF_RTIMER_TIMER, NRF_TIMER_CC_CHANNEL3);
+  timestamps.end = nrf_timer_cc_get(NRF_RTIMER_TIMER, NRF_TIMER_CC_CHANNEL2);
   timestamps.mpdu_duration = rx_buf.phr * BYTE_DURATION_RTIMER;
 
+
+  /* Use NRF_RTIMER_TIMER CC0 for conversion to rtimer(based on RTC) */
+  /* timestamps.sfd will be used in TSCH Module for slot calculation */
+  NRF_RTIMER_TIMER->TASKS_CAPTURE[0] = 1;
+  uint64_t now = RTIMER_NOW();
+  timestamps.framestart = now - ((NRF_RTIMER_TIMER->CC[0] - timestamps.framestart) / (float)62500 * RTIMER_SECOND);
+  timestamps.end = now - ((NRF_RTIMER_TIMER->CC[0] - timestamps.end) / (float)62500 * RTIMER_SECOND);
```

Comment:
You should be able to get a floating point constant by appending an `f` instead of having to do a cast.

I don't know anything about the code here, but is the floating point division standard practice for timer code?

Reply:
This is a conversion and has to be as exact a possible, 
but good point, there's a better way to do this calc without the float cast ;-)
```
timestamps.framestart = now - ((NRF_RTIMER_TIMER->CC[0] - timestamps.framestart) * RTIMER_SECOND / 62500 );
```